python/ray/data/_internal/split.py

```diff
@@ -90,7 +91,7 @@ def _split_single_block(
     meta: BlockMetadata,
     block_row: int,
     split_indices: List[int],
-) -> Tuple[int, List[Tuple[ObjectRef[Block], BlockMetadata]]]:
+) -> Tuple[int, List[Tuple[ObjectRef[Block], BlockMetadata, int]]]:
```

Comment:
we already calculate number of rows before in `_calculate_blocks_rows()`, can't we update `BlockMetadata.num_rows` in that function, so we can directly use `BlockMetadata.num_rows` instead of passing a separate argument?

Reply:
for `List[Tuple[ObjectRef[Block], BlockMetadata`, we can use type [BlockPartition](https://github.com/ray-project/ray/blob/master/python/ray/data/block.py#L135) here.